subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/DependencyHandlerScope.kt

```diff
@@ -47,11 +45,6 @@ private constructor(
     override val delegate: DependencyHandler
         get() = dependencies
 
-    @Deprecated(replaceWith = ReplaceWith("constraints"), message = "This method shouldn't be called because the most specific variant should be preferred by the Kotlin compiler", level = DeprecationLevel.HIDDEN)
```

Comment:
I think with `Deprecationlevel.HIDDEN` older plugins can still call this method, but new plugins can never access it.

That means we wouldn't have to remove this, but I'm not sure if that's worth it @eskatos wdyt?

Reply:
It was deprecated in Feb 2020, so it's been a while...

Reply:
Yes, `Deprecationlevel.HIDDEN` works as you described Sterling.

This was introduced in https://github.com/gradle/gradle/commit/ac16c2e02fe4f2b311b4062e5187349b34fa7f3a as a workaround for wrong scoping.

I think it is fine to remove in a major :+1: